crates/control/src/cmd/serve.rs

```diff
@@ -29,10 +30,16 @@ async fn serve(listener: TcpListener) -> anyhow::Result<()> {
     let (put_builds, fetch_builds) = init_builds_root(&config::settings().builds_root)?;
     let ctx = AppContext::new(db, put_builds, fetch_builds);
 
-    let server = startup::run(listener, ctx)?;
+    // TODO(johnny): For now, we run the API server and builder daemon together.
+    // We'll probably want to separate and independently deploy & scale these.
+    let server = startup::run(listener, ctx.clone())?.map_err(Into::into);
 
-    // The server runs until it receives a shutdown signal.
-    server.await?;
+    let builder_daemon =
+        crate::services::builder::serve_builds(ctx.clone(), shutdown::signal()).map_err(Into::into);
```

Comment:
We probably need to do better error handling here. Right now, an error encountered by the `builder` will stop the entire server. There are a lot of errors that can bubble up to this point and having the server exit is not really a great way to handle those. This is doubly bad if something is consistently wrong, as the builder will dequeue the problem Build and fail again, immediately. My local server is currently in a crashed state that I cannot get it out of. :grimacing:

Reply:
Okay, so I've tracked down what's causing the crash loop. We don't currently assert that the "local" BuildsRoot directory actually exists. We fail with a generic "file/directory does not exist" error when the Builder attempts to call `put_builds.put_build()`. I spent time looking for a bug with the temp directories we're making.

This isn't an issue that you introduced, but it is a footgun for someone trying to run a control plane build for the first time. At the very least, it's an easy reproduction of the crash loop problem I was talking about above.